### PR TITLE
Remove obsolete nodes from graph

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
Feel free to improve, my Python knowledge is still pretty basic.

I hope the workaround to remove useless final nodes is fine, but it would be good to test in different scenarios. I will soon publish a test version with working UI

### CODE REVIEW

**Comment by jirikuncar** [comment]
@lorenzo-cavazzi can you rebase on top #116?

File: app/blueprints/graph.py [hunk 1]
```diff
@@ -50,13 +50,15 @@
             rdfs:label ?target_label .
     ?activity rdfs:comment ?source_label .
     FILTER NOT EXISTS {{?activity rdf:type wfprov:WorkflowRun}}
+    FILTER EXISTS {{?activity rdf:type wfprov:ProcessRun}}
```
Comment (ableuler) [inline app/blueprints/graph.py]: Wasn't sure if you added those on purpose?

File: app/blueprints/graph.py [hunk 2]
```diff
@@ -50,13 +50,15 @@
...
     BIND (?entity AS ?target)
     BIND (?activity AS ?source)
   }} UNION {{
     ?activity prov:qualifiedUsage/prov:entity ?entity ;
               rdfs:comment ?target_label .
     ?entity rdfs:label ?source_label .
     FILTER NOT EXISTS {{?activity rdf:type wfprov:WorkflowRun}}
+    FILTER EXISTS {{?activity rdf:type wfprov:ProcessRun}}
```
Comment (ableuler) [inline app/blueprints/graph.py]: Same as above.

File: app/blueprints/graph.py (hunk 1 above)
Comment (jirikuncar) [reply app/blueprints/graph.py]: yes, I have added it on @lorenzo-cavazzi PC

**Comment by lorenzo-cavazzi** [comment]
Looks good.
I don't have the permission to merge it yet.

Btw there are still a few details to fix. Some graphs are good (see 1st image), others still have extra nodes (2nd image, we probably don't want those 2 `renku update` nodes on top).

Preview available at https://graphpreview.dev.renku.ch

Screenshot from 2019-04-04 09-05-56

Screenshot from 2019-04-04 09-05-29

**Pull request merged**